README.md

```diff
@@ -6,106 +8,295 @@
+
+The function that gets run is `<job class>.<queue name>`. So our example will
+trun a `MyClass.myQueue` function, passing the popped job. This function:
+
+- _must_ return a promise
+- _should_ ultimately return the `job.complete()` promise if successful
+
+When processing the job, any exceptions propagated out will result in the job
+being marked as `failed`.
+
 ```javascript
-// myproject/worker.js
-'use strict';
+// mymodule
+class MyClass {
 
-const qless = require('qless');
-qless.klassFinder.setModuleDir(__dirname + '/jobs');
-const client = new qless.Client();
-const worker = new qless.SerialWorker('myqueue', client);
-
-worker.run(err => {
-  console.log("ERROR IN WORKER: ", err);
-  // normally won't happen unless a serious (e.g. redis) error
-  // NOT triggered when a job [safely] fails
-});
+  // This method has two requirements:
+
+  static myQueue(job) {
+    return someAsynchronousWork()
+      .then(() => {
+        return someMoreAsynchronousWork();
+      })
+      .then(() => {
+        return job.complete();
+      })
+  }
+}
 ```
 
-To enable debugging, run with:
+As for running these jobs, `qless-js` comes with a binary that will continuously
+pop and run jobs:
 
 ```bash
-DEBUG='qless:*' node worker.js
+qless-js-worker \
+  --host='redis://my-redis-host:6379/' \
+  --queue=myQueue
 ```
 
-#### Example job
+## Job Class Importing
+
+The importing logic tries to `require` the klass string provided the job config.
+Failing that, it will strip off the last path segment (`/`-separated), and try
+importing that, treating the last segment as a property name of the imported
+entity. This continues until there's a successful `require` or until there is
+nothing that can be imported, in which case an exception is raised.
+
+Put another way, Split the provided name by `/`, and import the longest working
+prefix. The remaining components will be treated as attributes off of the
+import.
+
+For example, consider 'some/package/file.js/property/class'. We would traverse
+this as:
+
+- import 'some/package/file.js/property/class' (failed)
+- import 'some/package/file.js/property', get 'class' attribute (failed)
+- import 'some/package/file.js', get 'property.class' attribute (success)
+
+Technically, the targeted entity doesn't have to be a class. That's merely a
+convention inherited from other bindings. It must merely be an object with a
+property `<queue name>` which is a function that takes a job argument an returns
+a promise. For example, this would be a completely valid structure:
 
 ```javascript
-// myproject/jobs/MyJob.js
 module.exports = {
-  perform(job, cb) {
-    console.log(job.data.foo);
-    cb();
-    // to fail:
-    // cb('my error');
-    // job.fail('my error')'; cb();
-  }
+  queueOne: (job) => { ... },
+  queueTwo: (job) => { ... },
 }
 ```
 
-#### Example job with co/generators
+__Note__: A job class cannot be a path, by default. There is a [to-do](#to-dos)
```

Comment:
WRT the leaving the `node_modules` directory, we defend against that by first `Path.resovle`-ing the provided `klass` and then checking if that begins with a `.` or `/`, and there's a test case about exactly the example you provided.

As for how this will work in an application, take this with a grain of salt as I don't have a lot of experience with deploying node applications. I imagine that in some form or another one will have a sandboxed environment (I imagine / hope there's something akin to `virtualenv` or `bundle` in the Node community -- I don't know yet if `yarn` meets those criteria, but that's what I'd been using).

And in that environment, I imagine application code and `qless-js` are installed and readily importable. That you'd make jobs with a `klass` like `my-application/job` or the like, and you just run a `qless-js-worker` in upstart in the context of that environment?

My intention is to do as is done in the Node community -- the above doesn't necessarily express a die-hard opinion, but more what I have in mind with the understanding that there may be some gaps in my understanding :-)